File: vepweb/lib/model/company/Company.dart
part of vep.model.company;

class Company {
  String canonical;
  String name;
  String address;
  bool isVep;
  String content;

  CompanyForm toCompanyForm() {
    var companyForm = new CompanyForm();
    companyForm.canonical = canonical;
    companyForm.name = name;
    companyForm.address = address;
    companyForm.content = content;
    companyForm.isVep = isVep;
    return companyForm;
  }
}

class CompanyForm {
  @jsonIgnore
  String canonical;
  String name;
  String address;
  bool isVep;
  String content;
}

File: src/main/scala/vep/app/production/company/Company.scala
package vep.app.production.company

import scalikejdbc.WrappedResultSet
import spray.json.RootJsonFormat
import vep.framework.utils.JsonProtocol

case class Company(
  id: String,
  name: String,
  address: String,
  isVep: Boolean,
  content: String
)

object Company {
  import JsonProtocol._

  implicit val companyFormat: RootJsonFormat[Company] = jsonFormat5(Company.apply)

  def apply(resultSet: WrappedResultSet): Company = {
    new Company(
      id = resultSet.string("id"),
      name = resultSet.string("name"),
      address = resultSet.string("address"),
      isVep = resultSet.boolean("isVep"),
      content = resultSet.string("content"),
    )
  }
}

case class CompanyCreation(
  name: String,
  address: String,
  isVep: Boolean,
  content: String
)

object CompanyCreation {
  import JsonProtocol._

  implicit val companyCreationFormat: RootJsonFormat[CompanyCreation] = jsonFormat4(CompanyCreation.apply)
}